src/Authentication/Common.Authentication/Models/AzureRMProfile.cs

```diff
@@ -0,0 +1,140 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using Newtonsoft.Json;
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.Azure.Common.Authentication.Models
+{
+    /// <summary>
+    /// Represents Azure Resource Manager profile structure with default context, environments and token cache.
+    /// </summary>
+    [Serializable]
+    public sealed class AzureRMProfile : IAzureProfile
+    {
+        /// <summary>
+        /// Gets or sets Azure environments.
+        /// </summary>
+        public Dictionary<string, AzureEnvironment> Environments { get; set; }
+
+        /// <summary>
+        /// Gets or sets the token cache contents.
+        /// </summary>
+        public byte[] TokenCache { get; set; }
+
+        /// <summary>
+        /// Gets or sets the default azure context object.
+        /// </summary>
+        public AzureContext DefaultContext { get; set; }
+
+        /// <summary>
+        /// Gets the path of the profile file. 
+        /// </summary>
+        [JsonIgnore]
+        public string ProfilePath { get; private set; }
+
+        private void Load(string path)
+        {
+            this.ProfilePath = path;
+
+            if (!AzureSession.DataStore.DirectoryExists(AzureSession.ProfileDirectory))
+            {
+                AzureSession.DataStore.CreateDirectory(AzureSession.ProfileDirectory);
+            }
+
+            if (AzureSession.DataStore.FileExists(ProfilePath))
+            {
+                string contents = AzureSession.DataStore.ReadFileAsText(ProfilePath);
+                AzureRMProfile profile = JsonConvert.DeserializeObject<AzureRMProfile>(contents);
```

Comment:
Can the profile be null at this point?  If so, we should put in a null check here and throw an error message latting the user know we were unable to deserialize the profile.